static/js/topic_list_data.js

```diff
@@ -79,16 +79,18 @@ exports.get_list_info = function (stream_id, zoomed) {
             // same code we do when zoomed.
         }
 
-        const topic_info = {
-            topic_name,
-            unread: num_unread,
-            is_zero: num_unread === 0,
-            is_muted: is_topic_muted,
-            is_active_topic,
-            url: hash_util.by_stream_topic_uri(stream_id, topic_name),
-        };
-
-        items.push(topic_info);
+        if (!filter || topic_name.includes(filter)) {
```

Comment:
We probably want to be case-insensitive here, probably.  And we may eventually want something fancier that breaks topics into words if the filter is multi-word.  For now it might be worth extracting a really simply function in here called `function matches_filter(topic) {topic_name.toLowerCase().includes(topic)}` so that you can add a TODO comment in it.